nntrainer/src/neuralnet.cpp

```diff
@@ -346,13 +348,69 @@ int NeuralNetwork::init() {
       conv2d_layer->setWeightDecay(weight_decay);
       NN_INI_RETURN_STATUS();
 
-      status = conv2d_layer->setOptimizer(opt);
+      status = conv2d_layer->initialize(last);
       NN_INI_RETURN_STATUS();
 
-      status = conv2d_layer->initialize(last);
+      status = conv2d_layer->setOptimizer(opt);
       NN_INI_RETURN_STATUS();
+
       layers.push_back(conv2d_layer);
     } break;
+
+    case LAYER_POOLING2D: {
+      int size[POOLING2D_DIM];
+      std::shared_ptr<Pooling2DLayer> pooling2d_layer =
+        std::make_shared<Pooling2DLayer>();
+
+      pooling2d_layer->setInputDimension(previous_dim);
+
+      status =
+        getValues(POOLING2D_DIM,
+                  iniparser_getstring(
+                    ini, (layer_name + ":pooling_size").c_str(), unknown),
+                  (int *)size);
+
+      NN_INI_RETURN_STATUS();
+      status =
+        pooling2d_layer->setSize(size, Layer::PropertyType::pooling_size);
+
+      NN_INI_RETURN_STATUS();
+      status = getValues(
+        POOLING2D_DIM,
+        iniparser_getstring(ini, (layer_name + ":stride").c_str(), unknown),
+        (int *)size);
+      NN_INI_RETURN_STATUS();
+      status = pooling2d_layer->setSize(size, Layer::PropertyType::stride);
+      NN_INI_RETURN_STATUS();
+      status = getValues(
+        POOLING2D_DIM,
+        iniparser_getstring(ini, (layer_name + ":padding").c_str(), unknown),
+        (int *)size);
+      NN_INI_RETURN_STATUS();
+      status = pooling2d_layer->setSize(size, Layer::PropertyType::padding);
+      NN_INI_RETURN_STATUS();
+
+      pooling2d_layer->setPoolingType(
+        (nntrainer::Pooling2DLayer::PoolingType)parseType(
+          iniparser_getstring(ini, (layer_name + ":pooling").c_str(), unknown),
+          TOKEN_POOLING));
+
+      status = pooling2d_layer->initialize(last);
+      NN_INI_RETURN_STATUS();
+      layers.push_back(pooling2d_layer);
+    } break;
+
+    case LAYER_FLATTEN: {
+      std::shared_ptr<FlattenLayer> flatten_layer =
```

Comment:
replace with 
```
status = initFlattenLayer();
NN_INI_RETURN_STATUS();
```

Reply:
But at this time, we do not know the layers[position]->getOuptutDimension() because this is the independent layer.

Reply:
We know that this will not be the first layer. Hence, `layers[position]->getOutputDimension()` will exist as with `initFlattenLayer()`, position is set to be the last layer in layers.

Reply:
Got it.